packages/@react-stately/virtualizer/src/Virtualizer.ts

```diff
@@ -460,10 +460,6 @@ export class Virtualizer<T extends object, V, W> {
     }
 
     this._invalidationContext = context;
-    this._relayoutRaf = requestAnimationFrame(() => {
```

Comment:
This is equivalent to being called in afterRender because React does renders on rafs, so afterRender, which is called by a useLayoutEffect would be called at roughly the equivalent time, just in sync with React lifecycles now?

I noticed that we no longer need the _relayoutRaf at all. This was the only line that set it to anything other than null. As a result, relayoutNow needs some updates to remove unused code. And can remove willUnmount. 

processTransaction's move is similar to the changes I made to useValueEffect for strictmode? we depend on the previous render cycle? I think it's in an if/else with relayoutNow because processTransactionQueue will already call relayoutNow? so this avoids calling it a second time?

I think the comment on line 456 is now stale, it should be coalesce multiple relayouts in the same render cycle or something like that?

weird question, but would React 18 state batching make relayout unneeded, as relayouNow could be called directly?

Reply:
> This is equivalent to being called in afterRender because React does renders on rafs, so afterRender, which is called by a useLayoutEffect would be called at roughly the equivalent time, just in sync with React lifecycles now?

Close. useEffect runs asynchronously (not sure if in a raf or something else). useLayoutEffect happens synchronously, so actually this occurs sooner now. Might actually help with some other issues like flickering when things resize.

> I noticed that we no longer need the _relayoutRaf at all. This was the only line that set it to anything other than null. As a result, relayoutNow needs some updates to remove unused code. And can remove willUnmount.

possibly. I'm not actually sure if we will need to bring it back or not. If you were to call `relayout` from outside render, currently nothing would ever trigger a re-render. So we might need a way to detect this. I didn't see anything in the tests but still wondering.

> processTransaction's move is similar to the changes I made to useValueEffect for strictmode? we depend on the previous render cycle? 

yes.

> I think it's in an if/else with relayoutNow because processTransactionQueue will already call relayoutNow? so this avoids calling it a second time?

yes.

> weird question, but would React 18 state batching make relayout unneeded, as relayouNow could be called directly?

relayoutNow runs immediately, so there would be multiple relayouts even if the _result_ is batched. So I think relayout is still useful as a queuing mechanism.